install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -12,262 +12,728 @@ toc::[]
+capacity.
+
+Multiple persistent volume claims (PVCs) within the same project can bind to the
+same PV. However, once a PVC binds to a PV, that PV cannot be bound by a claim
+outside of the first claim's project. If the underlying storage needs to be
+accessed by multiple projects, then each project needs its own PV, which can
+point to the same physical storage. In this sense, a bound PV is tied to a
+project. For a detailed PV and PVC example, see the guide for
+https://github.com/openshift/origin/tree/master/examples/wordpress[WordPress and
+MySQL using NFS].
+
+For the cluster administrator, granting pods access to PVs involves:
+
+- knowing the group ID and/or user ID assigned to the actual storage,
+- understanding SELinux considerations, and
+- ensuring that these IDs are allowed in the range of legal IDs defined for the
+project and/or the SCC that matches the requirements of the pod.
+
+Group IDs, the user ID, and SELinux values are defined in the
+`*SecurityContext*` section in a pod definition. Group IDs are global to the pod
+and apply to all containers defined in the pod. User IDs can also be global, or
+specific to each container. Four sections control access to volumes:
+
+- link:#supplemental-groups[`*supplementalGroups*`]
+- link:#fsgroup[`*fsGroup*`]
+- link:#user-id[`*runAsUser*`]
+- link:#selinuxoptions[`*seLinuxOptions*`]
+
+[[sccs-defaults-allowed-ranges]]
+== SCCs, Defaults, and Allowed Ranges
+
+SCCs influence whether or not a pod is given a default user ID, `*fsGroup*` ID,
+supplemental group ID, and SELinux label. They also influence whether or not IDs
+supplied in the pod definition (or in the image) will be validated against a
+range of allowable IDs. If validation is required and fails, then the pod will
+also fail.
+
+SCCs define strategies, such as `*runAsUser*`, `*supplementalGroups*`, and
+`*fsGroup*`. These strategies are used to help decide if the pod is authorized.
+Strategy values set to *RunAsAny* are essentially stating that the pod can do
+what it wants regarding that strategy. Authorization is skipped for that
+strategy and no OpenShift default is produced based on that strategy. Therefore,
+IDs and SELinux labels in the resulting container are based on container
+defaults instead of OpenShift policies.
+
+For a quick summary of *RunAsAny*:
+
+- Any ID defined in the pod definition (or image) is allowed.
+- Absence of an ID in the pod definition (and in the image) results in the
+container assigning an ID, which is *root* (0) for Docker.
+- No SELinux labels are defined, so Docker will assign a unique (and generally not
+useful) label.
+
+For these reasons, SCCs with *RunAsAny* for ID-related strategies should be
+protected so that ordinary developers do not have access to the SCC. On the
+other hand, SCC strategies set to *MustRunAs* or *MustRunAsRange* trigger ID
+validation (for ID related strategies), and cause default values to be supplied
```

Comment:
s/ID related strategies/ID-related strategies  (as seen in the previous sentence)